packages/next/server/web/spec-extension/request.ts

```diff
@@ -20,14 +20,14 @@ export class NextRequest extends Request {
   }
 
   constructor(input: Request | string, init: RequestInit = {}) {
-    super(input, init)
-
+    const url = typeof input === 'string' ? input : input.url
+    super(validateURL(url), init)
```

Comment:
**question:** why are we limiting the Request constructor to strings only?

[Request.constructor()](https://developer.mozilla.org/en-US/docs/Web/API/Request/Request#syntax) supports either string or object. Passing an object would create a copy (with a couple of tweaks as explained in the doc).

With this change, we're limiting users to only the first possibility. Is that intended, and what's the reasoning behind?

Reply:
The reason is to warn users if they introduce a relative URL

Reply:
Thanks!
Then we should pass the `input` parameter to super constructor, and do the validation independently.
```js
const url = typeof input === 'string' ? input : input.url
validateURL(url)
super(input, init)
```

This way, `NextRequest` constructor will behave the same as `Response`, and provides extra checks on incoming urls.